From 57b2c4fe7caabc5519d19bcd715709673870bf3e Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 13 Jun 2017 17:15:45 +0900
Subject: [PATCH 1/2] Add a preference to force enable touch events without APZ

Firefox enables W3C touch events only when e10s is enabled since it's
requied by APZ (Asynchronous Pan & Zoom), but they are still meaningful
on touch devices even if e10s is disabled.

This patch adds a preference to enable them wihout e10s.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 widget/gtk/nsWindow.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index 2c8baeb3f4e8..987e17d7a952 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -428,6 +428,9 @@ nsWindow::nsWindow() {
 
 #if GTK_CHECK_VERSION(3, 4, 0)
   mLastScrollEventTime = GDK_CURRENT_TIME;
+
+  if (Preferences::GetInt("dom.w3c_touch_events.enabled", 0) == 3)
+    RegisterTouchWindow();
 #endif
   mPendingConfigures = 0;
   mCSDSupportLevel = CSD_SUPPORT_NONE;
-- 
2.17.1

